src/plugins/gui/providers/simplexProvider.ts

```diff
@@ -120,7 +120,7 @@ const SIMPLEX_ID_MAP: { [pluginId: string]: { [currencyCode: string]: string } }
 
 const allowedCurrencyCodes: FiatProviderAssetMap = { crypto: {}, fiat: {} }
 const allowedCountryCodes: { [code: string]: boolean } = {}
-const allowedPaymentTypes = { applepay: true, credit: true, googlepay: false }
+const allowedPaymentTypes: { [payment: string]: boolean } = { applepay: true, credit: true, googlepay: false }
```

Comment:
This type annotation stomps away useful information, since it prevents TypeScript from validating the key. It would be better to delete the type annotation and fix the revealed errors (adding in 'iach' as required).

Otherwise, if you really want a type annotation here, you could type it as `{[K in FiatPaymentType]: boolean}`, which will define one property per enum member.

Or, if the property is optional, you can do `{[K in FiatPaymentType]?: boolean}` instead.

Reply:
fixed

Reply:
Although I fixed using `{[payment: FiatPaymentType] boolean}` syntax. Is that effectively an equivalent?

Reply:
We found the issue: it was the `asEither` above. Fixing that causes the `in` syntax to work again.